$NetBSD: patch-ac,v 1.1 2006/11/05 18:46:58 rillig Exp $

SunPro sees an overloading ambiguity between operator bool() and
operator const char *().

--- permstr.cc.orig	2001-09-06 20:30:10.000000000 +0200
+++ permstr.cc	2006-11-05 19:12:50.020399608 +0100
@@ -183,8 +183,8 @@ PermString::static_initialize()
 bool
 operator==(PermString a, const char *b)
 {
-  if (!a || !b)
-    return !a && !b;
+  if (a == false || !b)
+    return (a == false) && !b;
   int l = strlen(b);
   return a.length() == l && memcmp(a.cc(), b, l) == 0;
 }
